// @ant-design/web3/dist/lib/connect-modal/components/LinkPanel.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = _interopRequireWildcard(require("react"));
var _antd = require("antd");
var _context = require("../context");
var _MainPanelHeader = _interopRequireDefault(require("./MainPanelHeader"));
var _WalletIcon = _interopRequireDefault(require("./WalletIcon"));
var _jsxRuntime = require("react/jsx-runtime");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
const LinkPanel = () => {
  const {
    connecting,
    updatePanelRoute,
    localeMessage,
    prefixCls,
    selectedWallet
  } = (0, _react.useContext)(_context.connectModalContext);
  _react.default.useEffect(() => {
    if (!connecting) {
      updatePanelRoute('init', true);
    }
  }, [connecting]);
  return /*#__PURE__*/(0, _jsxRuntime.jsxs)(_jsxRuntime.Fragment, {
    children: [/*#__PURE__*/(0, _jsxRuntime.jsx)(_MainPanelHeader.default, {
      title: `${localeMessage.linkWallet} ${selectedWallet.name}`
    }), /*#__PURE__*/(0, _jsxRuntime.jsxs)("div", {
      className: `${prefixCls}-link-panel`,
      children: [/*#__PURE__*/(0, _jsxRuntime.jsxs)("div", {
        className: `${prefixCls}-ripple-container`,
        children: [/*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
          className: `${prefixCls}-ripple`
        }), /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
          className: `${prefixCls}-ripple`
        }), /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
          className: `${prefixCls}-ripple`
        }), /*#__PURE__*/(0, _jsxRuntime.jsx)(_antd.Avatar, {
          size: 56,
          icon: /*#__PURE__*/(0, _jsxRuntime.jsx)(_WalletIcon.default, {
            wallet: selectedWallet
          })
        })]
      }), /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
        className: `${prefixCls}-wallet-connecting`,
        children: localeMessage.walletConnecting
      })]
    })]
  });
};
var _default = exports.default = LinkPanel;
