// @ant-design/web3/dist/esm/pay-panel/index.js
import React, { useContext, useState } from 'react';
import { ConfigProvider } from 'antd';
import classNames from 'classnames';
import { ChainList } from "./ChainList";
import { PayPanelContext } from "./PayPanelContext";
import { ShowCode } from "./ShowCode";
import { useStyle } from "./style";
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
export const PayPanel = props => {
  const {
    getPrefixCls
  } = useContext(ConfigProvider.ConfigContext);
  const prefixCls = getPrefixCls('web3-pay-panel');
  const {
    wrapSSR,
    hashId
  } = useStyle(prefixCls);
  const [selectedChainId, setSelectedChainId] = useState();
  const {
    amount,
    target,
    supportedChains,
    token,
    wallets,
    onFinish
  } = props;
  return wrapSSR( /*#__PURE__*/_jsx(PayPanelContext.Provider, {
    value: {
      amount,
      target,
      supportedChains,
      token,
      wallets,
      onFinish
    },
    children: /*#__PURE__*/_jsx("div", {
      className: classNames(prefixCls, hashId),
      children: /*#__PURE__*/_jsxs("div", {
        className: `${prefixCls}-content`,
        children: [!selectedChainId && /*#__PURE__*/_jsx(ChainList, {
          onChainSelected: e => {
            setSelectedChainId(e);
          }
        }), selectedChainId && /*#__PURE__*/_jsx(ShowCode, {
          selectedChainId: selectedChainId,
          onReturn: () => {
            setSelectedChainId(undefined);
          }
        })]
      })
    })
  }));
};
