// @ant-design/web3-assets/dist/lib/wallets/meta-mask.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.metadata_MetaMask = void 0;
var _web3Common = require("@ant-design/web3-common");
var _web3Icons = require("@ant-design/web3-icons");
var _jsxRuntime = require("react/jsx-runtime");
const metadata_MetaMask = exports.metadata_MetaMask = {
  icon: /*#__PURE__*/(0, _jsxRuntime.jsx)(_web3Icons.MetaMaskColorful, {}),
  name: 'MetaMask',
  remark: 'MetaMask Wallet',
  app: {
    link: 'https://metamask.io/'
  },
  extensions: [{
    key: 'Chrome',
    browserIcon: /*#__PURE__*/(0, _jsxRuntime.jsx)(_web3Icons.ChromeCircleColorful, {}),
    browserName: 'Chrome',
    link: 'https://chrome.google.com/webstore/detail/metamask/nkbihfbeogaeaoehlefnkodbefgpgknn',
    description: 'Access your wallet right from your favorite web browser.'
  }],
  group: 'Popular',
  transferQRCodeFormatter: params => {
    const {
      toAddress,
      chainId,
      amount,
      tokenAddress = '',
      decimal
    } = params;
    if (tokenAddress) {
      return `ethereum:${tokenAddress}@${chainId}/transfer?address=${toAddress}&uint256=${Number(amount) * Math.pow(10, 18 - decimal)}`;
    } else {
      return `ethereum:${toAddress}@${chainId}?value=${amount}`;
    }
  },
  supportChainTypes: [_web3Common.ChainType.EVM]
};
