// @ant-design/web3-assets/dist/lib/solana/chains.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SolanaTestnet = exports.SolanaDevnet = exports.Solana = void 0;
var _web3Common = require("@ant-design/web3-common");
var _web3Icons = require("@ant-design/web3-icons");
var _jsxRuntime = require("react/jsx-runtime");
const Solana = exports.Solana = {
  id: _web3Common.SolanaChainIds.MainnetBeta,
  name: 'Solana',
  icon: /*#__PURE__*/(0, _jsxRuntime.jsx)(_web3Icons.SolanaColorful, {}),
  browser: {
    icon: /*#__PURE__*/(0, _jsxRuntime.jsx)(_web3Icons.SolanaColorful, {}),
    getBrowserLink: (0, _web3Common.createGetBrowserLink)('https://explorer.solana.com')
  },
  nativeCurrency: {
    name: 'Solana',
    symbol: 'SOL',
    decimals: 9
  }
};
const SolanaDevnet = exports.SolanaDevnet = {
  id: _web3Common.SolanaChainIds.Devnet,
  name: 'Solana Devnet',
  icon: /*#__PURE__*/(0, _jsxRuntime.jsx)(_web3Icons.SolanaColorful, {}),
  browser: {
    icon: /*#__PURE__*/(0, _jsxRuntime.jsx)(_web3Icons.SolanaColorful, {}),
    getBrowserLink: (0, _web3Common.createGetBrowserLink)('https://explorer.solana.com/?cluster=devnet')
  },
  nativeCurrency: {
    name: 'Solana',
    symbol: 'SOL',
    decimals: 9
  }
};
const SolanaTestnet = exports.SolanaTestnet = {
  id: _web3Common.SolanaChainIds.Testnet,
  name: 'Solana Testnet',
  icon: /*#__PURE__*/(0, _jsxRuntime.jsx)(_web3Icons.SolanaColorful, {}),
  browser: {
    icon: /*#__PURE__*/(0, _jsxRuntime.jsx)(_web3Icons.SolanaColorful, {}),
    getBrowserLink: (0, _web3Common.createGetBrowserLink)('https://explorer.solana.com/?cluster=testnet')
  },
  nativeCurrency: {
    name: 'Solana',
    symbol: 'SOL',
    decimals: 9
  }
};
