// thirdweb/dist/esm/extensions/modular/ModularCore/write/installPublishedModule.js
import { getOrDeployInfraForPublishedContract } from "../../../../contract/deployment/utils/bootstrap.js";
import { installModule } from "../../__generated__/ModularCore/write/installModule.js";
/**
 * Install a published module on a modular contract
 * @param options - The options for installing a published module
 * @returns A prepared transaction to send
 * @extension MODULAR
 * @example
 * ```ts
 * import { installPublishedModule } from "thirdweb/modules/modular";
 * const transaction = installPublishedModule({
 *  client,
 *  chain,
 *  account,
 *  contract,
 *  moduleName: "MyModule",
 *  publisherAddress: "0x...",
 * });
 * await sendTransaction({ transaction, account });
 * ```
 */
export function installPublishedModule(options) {
    const { client, chain, account, contract, moduleName, constructorParams, publisherAddress, moduleData, } = options;
    return installModule({
        contract,
        asyncParams: async () => {
            const deployedModule = await getOrDeployInfraForPublishedContract({
                chain,
                client,
                account,
                contractId: moduleName,
                constructorParams: constructorParams || [],
                publisher: publisherAddress,
            });
            return {
                module: deployedModule.implementationContract.address,
                data: moduleData || "0x",
            };
        },
    });
}
