// thirdweb/dist/esm/extensions/modular/ModularCore/write/uninstallPublishedModule.js
import { getDeployedInfraContract } from "../../../../contract/deployment/utils/infra.js";
import { uninstallModule } from "../../__generated__/ModularCore/write/uninstallModule.js";
/**
 * Uninstall a published module
 * @param options - The options for uninstalling a published module
 * @returns A prepared transaction to send
 * @extension MODULAR
 * @example
 * ```ts
 * import { uninstallPublishedModule } from "thirdweb/modules/modular";
 * const transaction = uninstallPublishedModule({
 *  client,
 *  chain,
 *  contract,
 *  moduleName: "MyModule",
 *  publisherAddress: "0x...",
 * });
 * await sendTransaction({ transaction, account });
 * ```
 */
export function uninstallPublishedModule(options) {
    const { client, chain, contract, moduleName, publisherAddress, constructorParams, moduleData, } = options;
    return uninstallModule({
        contract,
        asyncParams: async () => {
            const deployedModule = await getDeployedInfraContract({
                chain,
                client,
                contractId: moduleName,
                constructorParams: constructorParams || [],
                publisher: publisherAddress,
            });
            return {
                module: deployedModule?.address,
                data: moduleData || "0x",
            };
        },
    });
}
