// @ant-design/web3-assets/dist/esm/chains/ethereum.js
import { ChainIds, ChainType, createGetBrowserLink } from '@ant-design/web3-common';
import { ArbitrumCircleColorful, AvalancheCircleColorful, BSCCircleColorful, EthereumCircleColorful, EthereumColorful, EthereumFilled, EtherscanCircleColorful, HardhatColorful, OkxWalletColorful, OptimismCircleColorful, PolygonCircleColorful, ScrollColorful } from '@ant-design/web3-icons';
import { jsx as _jsx } from "react/jsx-runtime";
export const Mainnet = {
  id: ChainIds.Mainnet,
  name: 'Ethereum',
  type: ChainType.EVM,
  icon: /*#__PURE__*/_jsx(EthereumCircleColorful, {}),
  browser: {
    icon: /*#__PURE__*/_jsx(EtherscanCircleColorful, {}),
    getBrowserLink: createGetBrowserLink('https://etherscan.io')
  },
  nativeCurrency: {
    icon: /*#__PURE__*/_jsx(EthereumFilled, {}),
    name: 'Ether',
    symbol: 'ETH',
    decimals: 18
  }
};

/**
 * @deprecated use sepolia or holesky instead
 */
export const Goerli = {
  id: ChainIds.Goerli,
  name: 'Goerli',
  type: ChainType.EVM,
  icon: /*#__PURE__*/_jsx(EthereumCircleColorful, {}),
  browser: {
    icon: /*#__PURE__*/_jsx(EthereumCircleColorful, {}),
    getBrowserLink: createGetBrowserLink('https://goerli.etherscan.io')
  },
  nativeCurrency: {
    name: 'Ether',
    symbol: 'ETH',
    decimals: 18
  }
};
export const Sepolia = {
  id: ChainIds.Sepolia,
  name: 'Sepolia',
  type: ChainType.EVM,
  icon: /*#__PURE__*/_jsx(EthereumCircleColorful, {}),
  browser: {
    icon: /*#__PURE__*/_jsx(EthereumCircleColorful, {}),
    getBrowserLink: createGetBrowserLink('https://sepolia.etherscan.io')
  },
  nativeCurrency: {
    name: 'Sepolia Ether',
    symbol: 'ETH',
    decimals: 18
  }
};
export const Holesky = {
  id: ChainIds.Holesky,
  name: 'Holesky',
  type: ChainType.EVM,
  icon: /*#__PURE__*/_jsx(EthereumCircleColorful, {}),
  browser: {
    icon: /*#__PURE__*/_jsx(EthereumCircleColorful, {}),
    getBrowserLink: createGetBrowserLink('https://holesky.etherscan.io')
  },
  nativeCurrency: {
    name: 'Holesky Ether',
    symbol: 'ETH',
    decimals: 18
  }
};
export const Polygon = {
  id: ChainIds.Polygon,
  name: 'Polygon',
  type: ChainType.EVM,
  icon: /*#__PURE__*/_jsx(PolygonCircleColorful, {}),
  browser: {
    icon: /*#__PURE__*/_jsx(PolygonCircleColorful, {}),
    getBrowserLink: createGetBrowserLink('https://polygonscan.com')
  },
  nativeCurrency: {
    name: 'MATIC',
    symbol: 'MATIC',
    decimals: 18
  }
};
export const BSC = {
  id: ChainIds.BSC,
  name: 'BNB Smart Chain',
  type: ChainType.EVM,
  icon: /*#__PURE__*/_jsx(BSCCircleColorful, {}),
  browser: {
    icon: /*#__PURE__*/_jsx(BSCCircleColorful, {}),
    getBrowserLink: createGetBrowserLink('https://bscscan.com')
  },
  nativeCurrency: {
    decimals: 18,
    name: 'BNB',
    symbol: 'BNB'
  }
};
export const Arbitrum = {
  id: ChainIds.Arbitrum,
  name: 'Arbitrum One',
  type: ChainType.EVM,
  icon: /*#__PURE__*/_jsx(ArbitrumCircleColorful, {}),
  browser: {
    icon: /*#__PURE__*/_jsx(ArbitrumCircleColorful, {}),
    getBrowserLink: createGetBrowserLink('https://arbiscan.io')
  },
  nativeCurrency: {
    name: 'Ether',
    symbol: 'ETH',
    decimals: 18
  }
};
export const Optimism = {
  id: ChainIds.Optimism,
  name: 'OP Mainnet',
  type: ChainType.EVM,
  icon: /*#__PURE__*/_jsx(OptimismCircleColorful, {}),
  browser: {
    icon: /*#__PURE__*/_jsx(OptimismCircleColorful, {}),
    getBrowserLink: createGetBrowserLink('https://optimistic.etherscan.io')
  },
  nativeCurrency: {
    name: 'Ether',
    symbol: 'ETH',
    decimals: 18
  }
};
export const Avalanche = {
  id: ChainIds.Avalanche,
  name: 'Avalanche',
  type: ChainType.EVM,
  icon: /*#__PURE__*/_jsx(AvalancheCircleColorful, {}),
  browser: {
    icon: /*#__PURE__*/_jsx(AvalancheCircleColorful, {}),
    getBrowserLink: createGetBrowserLink('https://snowtrace.io')
  },
  nativeCurrency: {
    name: 'Avalanche',
    symbol: 'AVAX',
    decimals: 18
  }
};

// OKX X1: https://www.okx.com/cn/x1/docs/developer/build-on-x1/quickstart
export const X1Testnet = {
  id: ChainIds.X1Testnet,
  name: 'X1 testnet',
  type: ChainType.EVM,
  icon: /*#__PURE__*/_jsx(OkxWalletColorful, {}),
  browser: {
    icon: /*#__PURE__*/_jsx(OkxWalletColorful, {}),
    getBrowserLink: createGetBrowserLink('https://www.okx.com/explorer/x1-test')
  },
  nativeCurrency: {
    name: 'OKB',
    symbol: 'OKB',
    decimals: 18
  }
};
export const Scroll = {
  id: ChainIds.Scroll,
  name: 'Scroll',
  type: ChainType.EVM,
  icon: /*#__PURE__*/_jsx(ScrollColorful, {}),
  browser: {
    icon: /*#__PURE__*/_jsx(ScrollColorful, {}),
    getBrowserLink: createGetBrowserLink('https://scrollscan.com/')
  },
  nativeCurrency: {
    name: 'Scroll ETH',
    symbol: 'ETH',
    decimals: 18
  }
};
export const ScrollSepolia = {
  id: ChainIds.ScrollSepolia,
  name: 'Scroll Sepolia Testnet',
  type: ChainType.EVM,
  icon: /*#__PURE__*/_jsx(ScrollColorful, {}),
  browser: {
    icon: /*#__PURE__*/_jsx(ScrollColorful, {}),
    getBrowserLink: createGetBrowserLink('https://sepolia.scrollscan.com/')
  },
  nativeCurrency: {
    name: 'Scroll ETH',
    symbol: 'ETH',
    decimals: 18
  }
};
export const Hardhat = {
  id: ChainIds.Hardhat,
  name: 'Hardhat',
  type: ChainType.EVM,
  icon: /*#__PURE__*/_jsx(HardhatColorful, {}),
  nativeCurrency: {
    icon: /*#__PURE__*/_jsx(EthereumFilled, {}),
    name: 'Ether',
    symbol: 'ETH',
    decimals: 18
  }
};
export const Localhost = {
  id: ChainIds.Localhost,
  name: 'Localhost',
  type: ChainType.EVM,
  icon: /*#__PURE__*/_jsx(EthereumColorful, {}),
  nativeCurrency: {
    icon: /*#__PURE__*/_jsx(EthereumFilled, {}),
    name: 'Ether',
    symbol: 'ETH',
    decimals: 18
  }
};
