// @ant-design/web3-common/dist/esm/types.js
export let ChainIds = /*#__PURE__*/function (ChainIds) {
  ChainIds[ChainIds["Mainnet"] = 1] = "Mainnet";
  ChainIds[ChainIds["Polygon"] = 137] = "Polygon";
  ChainIds[ChainIds["BSC"] = 56] = "BSC";
  ChainIds[ChainIds["Arbitrum"] = 42161] = "Arbitrum";
  ChainIds[ChainIds["Optimism"] = 10] = "Optimism";
  ChainIds[ChainIds["Goerli"] = 5] = "Goerli";
  ChainIds[ChainIds["Avalanche"] = 43114] = "Avalanche";
  ChainIds[ChainIds["X1Testnet"] = 195] = "X1Testnet";
  ChainIds[ChainIds["Sepolia"] = 11155111] = "Sepolia";
  ChainIds[ChainIds["Holesky"] = 17000] = "Holesky";
  ChainIds[ChainIds["Scroll"] = 534352] = "Scroll";
  ChainIds[ChainIds["ScrollSepolia"] = 534351] = "ScrollSepolia";
  ChainIds[ChainIds["Hardhat"] = 31337] = "Hardhat";
  ChainIds[ChainIds["Localhost"] = 1337] = "Localhost";
  return ChainIds;
}({});
export let SolanaChainIds = /*#__PURE__*/function (SolanaChainIds) {
  SolanaChainIds[SolanaChainIds["MainnetBeta"] = 2] = "MainnetBeta";
  SolanaChainIds[SolanaChainIds["Devnet"] = 3] = "Devnet";
  SolanaChainIds[SolanaChainIds["Testnet"] = 4] = "Testnet";
  return SolanaChainIds;
}({});
export let ChainType = /*#__PURE__*/function (ChainType) {
  ChainType["EVM"] = "EVM";
  ChainType["SVM"] = "SVM";
  ChainType["Bitcoin"] = "Bitcoin";
  return ChainType;
}({});

/**
 * @desc 浏览器扩展程序信息
 */

/**
 * @desc 钱包
 * @descEn Wallet
 */
