// @ant-design/web3-assets/dist/esm/wallets/meta-mask.js
import { ChainType } from '@ant-design/web3-common';
import { ChromeCircleColorful, MetaMaskColorful } from '@ant-design/web3-icons';
import { jsx as _jsx } from "react/jsx-runtime";
export const metadata_MetaMask = {
  icon: /*#__PURE__*/_jsx(MetaMaskColorful, {}),
  name: 'MetaMask',
  remark: 'MetaMask Wallet',
  app: {
    link: 'https://metamask.io/'
  },
  extensions: [{
    key: 'Chrome',
    browserIcon: /*#__PURE__*/_jsx(ChromeCircleColorful, {}),
    browserName: 'Chrome',
    link: 'https://chrome.google.com/webstore/detail/metamask/nkbihfbeogaeaoehlefnkodbefgpgknn',
    description: 'Access your wallet right from your favorite web browser.'
  }],
  group: 'Popular',
  transferQRCodeFormatter: params => {
    const {
      toAddress,
      chainId,
      amount,
      tokenAddress = '',
      decimal
    } = params;
    if (tokenAddress) {
      return `ethereum:${tokenAddress}@${chainId}/transfer?address=${toAddress}&uint256=${Number(amount) * Math.pow(10, 18 - decimal)}`;
    } else {
      return `ethereum:${toAddress}@${chainId}?value=${amount}`;
    }
  },
  supportChainTypes: [ChainType.EVM]
};
