// @ant-design/web3/dist/lib/pay-panel/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PayPanel = void 0;
var _react = _interopRequireWildcard(require("react"));
var _antd = require("antd");
var _classnames = _interopRequireDefault(require("classnames"));
var _ChainList = require("./ChainList");
var _PayPanelContext = require("./PayPanelContext");
var _ShowCode = require("./ShowCode");
var _style = require("./style");
var _jsxRuntime = require("react/jsx-runtime");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
const PayPanel = props => {
  const {
    getPrefixCls
  } = (0, _react.useContext)(_antd.ConfigProvider.ConfigContext);
  const prefixCls = getPrefixCls('web3-pay-panel');
  const {
    wrapSSR,
    hashId
  } = (0, _style.useStyle)(prefixCls);
  const [selectedChainId, setSelectedChainId] = (0, _react.useState)();
  const {
    amount,
    target,
    supportedChains,
    token,
    wallets,
    onFinish
  } = props;
  return wrapSSR( /*#__PURE__*/(0, _jsxRuntime.jsx)(_PayPanelContext.PayPanelContext.Provider, {
    value: {
      amount,
      target,
      supportedChains,
      token,
      wallets,
      onFinish
    },
    children: /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
      className: (0, _classnames.default)(prefixCls, hashId),
      children: /*#__PURE__*/(0, _jsxRuntime.jsxs)("div", {
        className: `${prefixCls}-content`,
        children: [!selectedChainId && /*#__PURE__*/(0, _jsxRuntime.jsx)(_ChainList.ChainList, {
          onChainSelected: e => {
            setSelectedChainId(e);
          }
        }), selectedChainId && /*#__PURE__*/(0, _jsxRuntime.jsx)(_ShowCode.ShowCode, {
          selectedChainId: selectedChainId,
          onReturn: () => {
            setSelectedChainId(undefined);
          }
        })]
      })
    })
  }));
};
exports.PayPanel = PayPanel;
