// @ant-design/web3/dist/lib/crypto-input/style/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useCryptoInputStyle = useCryptoInputStyle;
exports.useStyle = useStyle;
var _classnames = _interopRequireDefault(require("classnames"));
var _useStyle = require("../../theme/useStyle");
var _constants = require("../constants");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
const getTokenStyle = token => {
  const {
    componentCls,
    antCls,
    paddingXS,
    padding,
    colorBorder,
    borderRadiusLG,
    fontWeightStrong,
    colorTextTertiary,
    marginSM,
    marginXS,
    colorInfoText,
    lineHeightSM,
    lineHeightHeading1,
    lineHeightHeading3
  } = token;
  const {
    inputFontSize,
    inputFontSizeSM,
    inputFontSizeLG,
    paddingInline,
    paddingBlock
  } = _constants.CRYPTO_INPUT_TOKEN;
  return {
    [`${componentCls}-wrapper`]: {
      paddingBlock: paddingXS,
      paddingInline: padding,
      border: `1px solid ${colorBorder}`,
      borderRadius: borderRadiusLG,
      rowGap: marginXS,
      width: '100%',
      boxSizing: 'border-box',
      [`${antCls}-input-number`]: {
        fontSize: inputFontSize,
        [`${antCls}-input-number-input`]: {
          fontWeight: fontWeightStrong,
          lineHeight: lineHeightHeading3,
          paddingInline: paddingInline,
          paddingBlock: paddingBlock,
          '&::placeholder': {
            fontWeight: 'normal'
          }
        }
      },
      [`${antCls}-input-number-lg`]: {
        fontSize: inputFontSizeLG,
        [`${antCls}-input-number-input`]: {
          lineHeight: lineHeightHeading1
        }
      },
      [`${antCls}-input-number-sm`]: {
        fontSize: inputFontSizeSM,
        [`${antCls}-input-number-input`]: {
          lineHeight: lineHeightSM
        }
      },
      [`${antCls}-input-number-group-addon`]: {
        paddingInlineEnd: paddingInline
      }
    },
    [`${componentCls}-amount`]: {
      flex: 1,
      border: 'none'
    },
    [`${componentCls}-footer`]: {
      [`${componentCls}-default-footer`]: {
        fontWeight: fontWeightStrong,
        [`${componentCls}-total-price`]: {
          flex: 1,
          color: colorTextTertiary,
          marginInlineEnd: marginSM
        },
        [`${componentCls}-token-balance`]: {
          color: colorTextTertiary,
          flex: '0 0 auto'
        },
        [`${componentCls}-max-button`]: {
          marginInlineStart: marginXS,
          color: colorInfoText
        }
      }
    }
  };
};
function useStyle(prefixCls) {
  return (0, _useStyle.useStyle)('crypto-input', token => {
    const proListToken = {
      ...token,
      componentCls: `.${prefixCls}`
    };
    return [getTokenStyle(proListToken)];
  });
}
function useCryptoInputStyle() {
  const prefixCls = 'web3-crypto-input';
  const {
    wrapSSR,
    hashId
  } = useStyle(prefixCls);
  const getClsName = (cls, ...extraCls) => (0, _classnames.default)(`${prefixCls}-${cls}`, hashId, ...extraCls);
  return {
    wrapSSR,
    getClsName
  };
}
