// @ant-design/web3-assets/dist/lib/chains/ethereum.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.X1Testnet = exports.Sepolia = exports.ScrollSepolia = exports.Scroll = exports.Polygon = exports.Optimism = exports.Mainnet = exports.Localhost = exports.Holesky = exports.Hardhat = exports.Goerli = exports.BSC = exports.Avalanche = exports.Arbitrum = void 0;
var _web3Common = require("@ant-design/web3-common");
var _web3Icons = require("@ant-design/web3-icons");
var _jsxRuntime = require("react/jsx-runtime");
const Mainnet = exports.Mainnet = {
  id: _web3Common.ChainIds.Mainnet,
  name: 'Ethereum',
  type: _web3Common.ChainType.EVM,
  icon: /*#__PURE__*/(0, _jsxRuntime.jsx)(_web3Icons.EthereumCircleColorful, {}),
  browser: {
    icon: /*#__PURE__*/(0, _jsxRuntime.jsx)(_web3Icons.EtherscanCircleColorful, {}),
    getBrowserLink: (0, _web3Common.createGetBrowserLink)('https://etherscan.io')
  },
  nativeCurrency: {
    icon: /*#__PURE__*/(0, _jsxRuntime.jsx)(_web3Icons.EthereumFilled, {}),
    name: 'Ether',
    symbol: 'ETH',
    decimals: 18
  }
};

/**
 * @deprecated use sepolia or holesky instead
 */
const Goerli = exports.Goerli = {
  id: _web3Common.ChainIds.Goerli,
  name: 'Goerli',
  type: _web3Common.ChainType.EVM,
  icon: /*#__PURE__*/(0, _jsxRuntime.jsx)(_web3Icons.EthereumCircleColorful, {}),
  browser: {
    icon: /*#__PURE__*/(0, _jsxRuntime.jsx)(_web3Icons.EthereumCircleColorful, {}),
    getBrowserLink: (0, _web3Common.createGetBrowserLink)('https://goerli.etherscan.io')
  },
  nativeCurrency: {
    name: 'Ether',
    symbol: 'ETH',
    decimals: 18
  }
};
const Sepolia = exports.Sepolia = {
  id: _web3Common.ChainIds.Sepolia,
  name: 'Sepolia',
  type: _web3Common.ChainType.EVM,
  icon: /*#__PURE__*/(0, _jsxRuntime.jsx)(_web3Icons.EthereumCircleColorful, {}),
  browser: {
    icon: /*#__PURE__*/(0, _jsxRuntime.jsx)(_web3Icons.EthereumCircleColorful, {}),
    getBrowserLink: (0, _web3Common.createGetBrowserLink)('https://sepolia.etherscan.io')
  },
  nativeCurrency: {
    name: 'Sepolia Ether',
    symbol: 'ETH',
    decimals: 18
  }
};
const Holesky = exports.Holesky = {
  id: _web3Common.ChainIds.Holesky,
  name: 'Holesky',
  type: _web3Common.ChainType.EVM,
  icon: /*#__PURE__*/(0, _jsxRuntime.jsx)(_web3Icons.EthereumCircleColorful, {}),
  browser: {
    icon: /*#__PURE__*/(0, _jsxRuntime.jsx)(_web3Icons.EthereumCircleColorful, {}),
    getBrowserLink: (0, _web3Common.createGetBrowserLink)('https://holesky.etherscan.io')
  },
  nativeCurrency: {
    name: 'Holesky Ether',
    symbol: 'ETH',
    decimals: 18
  }
};
const Polygon = exports.Polygon = {
  id: _web3Common.ChainIds.Polygon,
  name: 'Polygon',
  type: _web3Common.ChainType.EVM,
  icon: /*#__PURE__*/(0, _jsxRuntime.jsx)(_web3Icons.PolygonCircleColorful, {}),
  browser: {
    icon: /*#__PURE__*/(0, _jsxRuntime.jsx)(_web3Icons.PolygonCircleColorful, {}),
    getBrowserLink: (0, _web3Common.createGetBrowserLink)('https://polygonscan.com')
  },
  nativeCurrency: {
    name: 'MATIC',
    symbol: 'MATIC',
    decimals: 18
  }
};
const BSC = exports.BSC = {
  id: _web3Common.ChainIds.BSC,
  name: 'BNB Smart Chain',
  type: _web3Common.ChainType.EVM,
  icon: /*#__PURE__*/(0, _jsxRuntime.jsx)(_web3Icons.BSCCircleColorful, {}),
  browser: {
    icon: /*#__PURE__*/(0, _jsxRuntime.jsx)(_web3Icons.BSCCircleColorful, {}),
    getBrowserLink: (0, _web3Common.createGetBrowserLink)('https://bscscan.com')
  },
  nativeCurrency: {
    decimals: 18,
    name: 'BNB',
    symbol: 'BNB'
  }
};
const Arbitrum = exports.Arbitrum = {
  id: _web3Common.ChainIds.Arbitrum,
  name: 'Arbitrum One',
  type: _web3Common.ChainType.EVM,
  icon: /*#__PURE__*/(0, _jsxRuntime.jsx)(_web3Icons.ArbitrumCircleColorful, {}),
  browser: {
    icon: /*#__PURE__*/(0, _jsxRuntime.jsx)(_web3Icons.ArbitrumCircleColorful, {}),
    getBrowserLink: (0, _web3Common.createGetBrowserLink)('https://arbiscan.io')
  },
  nativeCurrency: {
    name: 'Ether',
    symbol: 'ETH',
    decimals: 18
  }
};
const Optimism = exports.Optimism = {
  id: _web3Common.ChainIds.Optimism,
  name: 'OP Mainnet',
  type: _web3Common.ChainType.EVM,
  icon: /*#__PURE__*/(0, _jsxRuntime.jsx)(_web3Icons.OptimismCircleColorful, {}),
  browser: {
    icon: /*#__PURE__*/(0, _jsxRuntime.jsx)(_web3Icons.OptimismCircleColorful, {}),
    getBrowserLink: (0, _web3Common.createGetBrowserLink)('https://optimistic.etherscan.io')
  },
  nativeCurrency: {
    name: 'Ether',
    symbol: 'ETH',
    decimals: 18
  }
};
const Avalanche = exports.Avalanche = {
  id: _web3Common.ChainIds.Avalanche,
  name: 'Avalanche',
  type: _web3Common.ChainType.EVM,
  icon: /*#__PURE__*/(0, _jsxRuntime.jsx)(_web3Icons.AvalancheCircleColorful, {}),
  browser: {
    icon: /*#__PURE__*/(0, _jsxRuntime.jsx)(_web3Icons.AvalancheCircleColorful, {}),
    getBrowserLink: (0, _web3Common.createGetBrowserLink)('https://snowtrace.io')
  },
  nativeCurrency: {
    name: 'Avalanche',
    symbol: 'AVAX',
    decimals: 18
  }
};

// OKX X1: https://www.okx.com/cn/x1/docs/developer/build-on-x1/quickstart
const X1Testnet = exports.X1Testnet = {
  id: _web3Common.ChainIds.X1Testnet,
  name: 'X1 testnet',
  type: _web3Common.ChainType.EVM,
  icon: /*#__PURE__*/(0, _jsxRuntime.jsx)(_web3Icons.OkxWalletColorful, {}),
  browser: {
    icon: /*#__PURE__*/(0, _jsxRuntime.jsx)(_web3Icons.OkxWalletColorful, {}),
    getBrowserLink: (0, _web3Common.createGetBrowserLink)('https://www.okx.com/explorer/x1-test')
  },
  nativeCurrency: {
    name: 'OKB',
    symbol: 'OKB',
    decimals: 18
  }
};
const Scroll = exports.Scroll = {
  id: _web3Common.ChainIds.Scroll,
  name: 'Scroll',
  type: _web3Common.ChainType.EVM,
  icon: /*#__PURE__*/(0, _jsxRuntime.jsx)(_web3Icons.ScrollColorful, {}),
  browser: {
    icon: /*#__PURE__*/(0, _jsxRuntime.jsx)(_web3Icons.ScrollColorful, {}),
    getBrowserLink: (0, _web3Common.createGetBrowserLink)('https://scrollscan.com/')
  },
  nativeCurrency: {
    name: 'Scroll ETH',
    symbol: 'ETH',
    decimals: 18
  }
};
const ScrollSepolia = exports.ScrollSepolia = {
  id: _web3Common.ChainIds.ScrollSepolia,
  name: 'Scroll Sepolia Testnet',
  type: _web3Common.ChainType.EVM,
  icon: /*#__PURE__*/(0, _jsxRuntime.jsx)(_web3Icons.ScrollColorful, {}),
  browser: {
    icon: /*#__PURE__*/(0, _jsxRuntime.jsx)(_web3Icons.ScrollColorful, {}),
    getBrowserLink: (0, _web3Common.createGetBrowserLink)('https://sepolia.scrollscan.com/')
  },
  nativeCurrency: {
    name: 'Scroll ETH',
    symbol: 'ETH',
    decimals: 18
  }
};
const Hardhat = exports.Hardhat = {
  id: _web3Common.ChainIds.Hardhat,
  name: 'Hardhat',
  type: _web3Common.ChainType.EVM,
  icon: /*#__PURE__*/(0, _jsxRuntime.jsx)(_web3Icons.HardhatColorful, {}),
  nativeCurrency: {
    icon: /*#__PURE__*/(0, _jsxRuntime.jsx)(_web3Icons.EthereumFilled, {}),
    name: 'Ether',
    symbol: 'ETH',
    decimals: 18
  }
};
const Localhost = exports.Localhost = {
  id: _web3Common.ChainIds.Localhost,
  name: 'Localhost',
  type: _web3Common.ChainType.EVM,
  icon: /*#__PURE__*/(0, _jsxRuntime.jsx)(_web3Icons.EthereumColorful, {}),
  nativeCurrency: {
    icon: /*#__PURE__*/(0, _jsxRuntime.jsx)(_web3Icons.EthereumFilled, {}),
    name: 'Ether',
    symbol: 'ETH',
    decimals: 18
  }
};
