// @ant-design/web3/dist/esm/connect-button/chain-select.js
import React, { useContext } from 'react';
import { DownOutlined } from '@ant-design/icons';
import { Button, ConfigProvider, Dropdown, Space } from 'antd';
import classNames from 'classnames';
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
export const ChainSelect = ({
  className,
  onSwitchChain,
  style,
  chains,
  hashId,
  currentChain,
  buttonProps
}) => {
  const {
    getPrefixCls
  } = useContext(ConfigProvider.ConfigContext);
  const prefixCls = getPrefixCls('web3-connect-button-chain-select');
  return /*#__PURE__*/_jsx(Button, {
    className: `${prefixCls}-button`,
    ...buttonProps,
    children: /*#__PURE__*/_jsx(Dropdown, {
      className: classNames(className, hashId, prefixCls),
      overlayClassName: `${prefixCls}-dropdown`,
      menu: {
        items: chains.map(chain => ({
          key: chain.id,
          label: chain.name,
          icon: chain.icon,
          onClick: () => {
            onSwitchChain?.(chain);
          }
        }))
      },
      trigger: ['click'],
      children: /*#__PURE__*/_jsx("div", {
        className: `${prefixCls}-placeholder`,
        style: style,
        children: /*#__PURE__*/_jsxs(Space, {
          children: [currentChain?.icon, currentChain?.name, /*#__PURE__*/_jsx(DownOutlined, {})]
        })
      })
    })
  });
};
ChainSelect.displayName = 'ChainSelect';
