// thirdweb/dist/esm/exports/extensions/modular.js
/**
 * READ
 */
export { getInstalledModules, isGetInstalledModulesSupported, } from "../../extensions/modular/__generated__/ModularCore/read/getInstalledModules.js";
export { getSupportedCallbackFunctions, isGetSupportedCallbackFunctionsSupported, } from "../../extensions/modular/__generated__/ModularCore/read/getSupportedCallbackFunctions.js";
export { hasAnyRole, } from "../../extensions/modular/__generated__/ModularCore/read/hasAnyRole.js";
export { hasAllRoles, } from "../../extensions/modular/__generated__/ModularCore/read/hasAllRoles.js";
export { owner } from "../../extensions/modular/__generated__/ModularCore/read/owner.js";
export { ownershipHandoverExpiresAt, } from "../../extensions/modular/__generated__/ModularCore/read/ownershipHandoverExpiresAt.js";
export { rolesOf, } from "../../extensions/modular/__generated__/ModularCore/read/rolesOf.js";
/**
 * Write
 */
export { cancelOwnershipHandover } from "../../extensions/modular/__generated__/ModularCore/write/cancelOwnershipHandover.js";
export { completeOwnershipHandover, } from "../../extensions/modular/__generated__/ModularCore/write/completeOwnershipHandover.js";
export { grantRoles, } from "../../extensions/modular/__generated__/ModularCore/write/grantRoles.js";
export { installModule, isInstallModuleSupported, } from "../../extensions/modular/__generated__/ModularCore/write/installModule.js";
export { renounceOwnership } from "../../extensions/modular/__generated__/ModularCore/write/renounceOwnership.js";
export { requestOwnershipHandover } from "../../extensions/modular/__generated__/ModularCore/write/requestOwnershipHandover.js";
export { renounceRoles, } from "../../extensions/modular/__generated__/ModularCore/write/renounceRoles.js";
export { revokeRoles, } from "../../extensions/modular/__generated__/ModularCore/write/revokeRoles.js";
export { transferOwnership, } from "../../extensions/modular/__generated__/ModularCore/write/transferOwnership.js";
export { uninstallModule, isUninstallModuleSupported, } from "../../extensions/modular/__generated__/ModularCore/write/uninstallModule.js";
export { getModuleConfig } from "../../extensions/modular/__generated__/ModularModule/read/getModuleConfig.js";
export { installPublishedModule, } from "../../extensions/modular/ModularCore/write/installPublishedModule.js";
export { uninstallPublishedModule, } from "../../extensions/modular/ModularCore/write/uninstallPublishedModule.js";
export { uninstallModuleByProxy, } from "../../extensions/modular/ModularCore/write/uninstallModuleByProxy.js";
