// thirdweb/dist/cjs/extensions/modular/ModularCore/write/uninstallPublishedModule.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.uninstallPublishedModule = uninstallPublishedModule;
const infra_js_1 = require("../../../../contract/deployment/utils/infra.js");
const uninstallModule_js_1 = require("../../__generated__/ModularCore/write/uninstallModule.js");
/**
 * Uninstall a published module
 * @param options - The options for uninstalling a published module
 * @returns A prepared transaction to send
 * @extension MODULAR
 * @example
 * ```ts
 * import { uninstallPublishedModule } from "thirdweb/modules/modular";
 * const transaction = uninstallPublishedModule({
 *  client,
 *  chain,
 *  contract,
 *  moduleName: "MyModule",
 *  publisherAddress: "0x...",
 * });
 * await sendTransaction({ transaction, account });
 * ```
 */
function uninstallPublishedModule(options) {
    const { client, chain, contract, moduleName, publisherAddress, constructorParams, moduleData, } = options;
    return (0, uninstallModule_js_1.uninstallModule)({
        contract,
        asyncParams: async () => {
            const deployedModule = await (0, infra_js_1.getDeployedInfraContract)({
                chain,
                client,
                contractId: moduleName,
                constructorParams: constructorParams || [],
                publisher: publisherAddress,
            });
            return {
                module: deployedModule?.address,
                data: moduleData || "0x",
            };
        },
    });
}
