// @ant-design/web3-assets/dist/esm/solana/chains.js
import { createGetBrowserLink, SolanaChainIds } from '@ant-design/web3-common';
import { SolanaColorful } from '@ant-design/web3-icons';
import { jsx as _jsx } from "react/jsx-runtime";
export const Solana = {
  id: SolanaChainIds.MainnetBeta,
  name: 'Solana',
  icon: /*#__PURE__*/_jsx(SolanaColorful, {}),
  browser: {
    icon: /*#__PURE__*/_jsx(SolanaColorful, {}),
    getBrowserLink: createGetBrowserLink('https://explorer.solana.com')
  },
  nativeCurrency: {
    name: 'Solana',
    symbol: 'SOL',
    decimals: 9
  }
};
export const SolanaDevnet = {
  id: SolanaChainIds.Devnet,
  name: 'Solana Devnet',
  icon: /*#__PURE__*/_jsx(SolanaColorful, {}),
  browser: {
    icon: /*#__PURE__*/_jsx(SolanaColorful, {}),
    getBrowserLink: createGetBrowserLink('https://explorer.solana.com/?cluster=devnet')
  },
  nativeCurrency: {
    name: 'Solana',
    symbol: 'SOL',
    decimals: 9
  }
};
export const SolanaTestnet = {
  id: SolanaChainIds.Testnet,
  name: 'Solana Testnet',
  icon: /*#__PURE__*/_jsx(SolanaColorful, {}),
  browser: {
    icon: /*#__PURE__*/_jsx(SolanaColorful, {}),
    getBrowserLink: createGetBrowserLink('https://explorer.solana.com/?cluster=testnet')
  },
  nativeCurrency: {
    name: 'Solana',
    symbol: 'SOL',
    decimals: 9
  }
};
