// @ant-design/web3/dist/esm/connect-modal/components/DefaultGuidePanel.js
import React, { useContext } from 'react';
import { Avatar, Button } from 'antd';
import classNames from 'classnames';
import { connectModalContext } from "../context";
import MainPanelHeader from "./MainPanelHeader";
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
import { Fragment as _Fragment } from "react/jsx-runtime";
const DefaultGuidePanel = props => {
  const {
    localeMessage,
    prefixCls
  } = useContext(connectModalContext);
  const defaultGuide = {
    title: localeMessage.guideTitle,
    infos: [{
      icon: 'https://mdn.alipayobjects.com/huamei_mutawc/afts/img/A*ApSUSaoUa_sAAAAAAAAAAAAADlrGAQ/original',
      title: localeMessage.guideInfos1Title,
      description: localeMessage.guideInfos1Desc
    }, {
      icon: 'https://mdn.alipayobjects.com/huamei_mutawc/afts/img/A*3lD7QpnbCPcAAAAAAAAAAAAADlrGAQ/original',
      title: localeMessage.guideInfos2Title,
      description: localeMessage.guideInfos2Desc
    }, {
      icon: 'https://mdn.alipayobjects.com/huamei_mutawc/afts/img/A*gTROQqEY_TcAAAAAAAAAAAAADlrGAQ/original',
      title: localeMessage.guideInfos3Title,
      description: localeMessage.guideInfos3Desc
    }],
    moreLink: 'https://ethereum.org/en/wallets/',
    getWalletBtnText: localeMessage.guideInfosGetWalletBtnText,
    moreLinkText: localeMessage.guideInfosMoreLinkText
  };
  const {
    guide = defaultGuide
  } = props;
  if (!guide) {
    return null;
  }
  const renderContent = () => {
    if ( /*#__PURE__*/React.isValidElement(guide)) {
      return guide;
    }
    const {
      title,
      infos
    } = guide;
    return /*#__PURE__*/_jsxs(_Fragment, {
      children: [props.simple && /*#__PURE__*/_jsx(MainPanelHeader, {}), /*#__PURE__*/_jsx("h2", {
        className: `${prefixCls}-guide-title`,
        children: title
      }), /*#__PURE__*/_jsx("div", {
        className: `${prefixCls}-guide-list`,
        children: infos.map(info => {
          return /*#__PURE__*/_jsxs("div", {
            className: `${prefixCls}-guide-item`,
            children: [typeof info.icon === 'string' ? /*#__PURE__*/_jsx(Avatar, {
              className: `${prefixCls}-guide-item-icon`,
              shape: "square",
              src: info.icon
            }) : info.icon, /*#__PURE__*/_jsxs("div", {
              className: `${prefixCls}-guide-item-content`,
              children: [/*#__PURE__*/_jsx("h3", {
                className: `${prefixCls}-guide-item-title`,
                children: info.title
              }), /*#__PURE__*/_jsx("div", {
                className: `${prefixCls}-guide-item-description`,
                children: info.description
              })]
            })]
          }, info.title?.toString());
        })
      })]
    });
  };
  const {
    moreLinkText = defaultGuide.moreLinkText
  } = guide;
  return /*#__PURE__*/_jsxs("div", {
    className: classNames(`${prefixCls}-guide-panel`, {
      [`${prefixCls}-simple-guide-panel`]: props.simple
    }),
    children: [renderContent(), /*#__PURE__*/_jsx(Button, {
      block: true,
      className: `${prefixCls}-more`,
      href: guide?.moreLink,
      target: "_blank",
      size: "large",
      children: moreLinkText
    })]
  });
};
export default DefaultGuidePanel;
