// @ant-design/web3/dist/esm/connect-modal/components/LinkPanel.js
import React, { useContext } from 'react';
import { Avatar } from 'antd';
import { connectModalContext } from "../context";
import MainPanelHeader from "./MainPanelHeader";
import WalletIcon from "./WalletIcon";
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
import { Fragment as _Fragment } from "react/jsx-runtime";
const LinkPanel = () => {
  const {
    connecting,
    updatePanelRoute,
    localeMessage,
    prefixCls,
    selectedWallet
  } = useContext(connectModalContext);
  React.useEffect(() => {
    if (!connecting) {
      updatePanelRoute('init', true);
    }
  }, [connecting]);
  return /*#__PURE__*/_jsxs(_Fragment, {
    children: [/*#__PURE__*/_jsx(MainPanelHeader, {
      title: `${localeMessage.linkWallet} ${selectedWallet.name}`
    }), /*#__PURE__*/_jsxs("div", {
      className: `${prefixCls}-link-panel`,
      children: [/*#__PURE__*/_jsxs("div", {
        className: `${prefixCls}-ripple-container`,
        children: [/*#__PURE__*/_jsx("div", {
          className: `${prefixCls}-ripple`
        }), /*#__PURE__*/_jsx("div", {
          className: `${prefixCls}-ripple`
        }), /*#__PURE__*/_jsx("div", {
          className: `${prefixCls}-ripple`
        }), /*#__PURE__*/_jsx(Avatar, {
          size: 56,
          icon: /*#__PURE__*/_jsx(WalletIcon, {
            wallet: selectedWallet
          })
        })]
      }), /*#__PURE__*/_jsx("div", {
        className: `${prefixCls}-wallet-connecting`,
        children: localeMessage.walletConnecting
      })]
    })]
  });
};
export default LinkPanel;
