// @ant-design/web3/dist/esm/connect-modal/components/MainPanel.js
import React, { useContext } from 'react';
import { connectModalContext } from "../context";
import DefaultGuidePanel from "./DefaultGuidePanel";
import LinkPanel from "./LinkPanel";
import QrCode from "./QrCode";
import WalletCard from "./WalletCard";
import { jsx as _jsx } from "react/jsx-runtime";
import { jsxs as _jsxs } from "react/jsx-runtime";
const MainPanel = props => {
  const {
    guide,
    simple
  } = props;
  const {
    prefixCls,
    panelRoute,
    selectedWallet
  } = useContext(connectModalContext);
  return /*#__PURE__*/_jsxs("div", {
    className: `${prefixCls}-main-panel`,
    children: [panelRoute === 'guide' || panelRoute === 'init' ? /*#__PURE__*/_jsx(DefaultGuidePanel, {
      guide: guide,
      simple: simple
    }) : null, panelRoute === 'wallet' && selectedWallet ? /*#__PURE__*/_jsx(WalletCard, {}) : null, panelRoute === 'link' && selectedWallet ? /*#__PURE__*/_jsx(LinkPanel, {}) : null, panelRoute === 'qrCode' && selectedWallet ? /*#__PURE__*/_jsx(QrCode, {
      wallet: selectedWallet,
      simple: simple
    }) : null, panelRoute === 'downloadQrCode' && selectedWallet ? /*#__PURE__*/_jsx(QrCode, {
      wallet: selectedWallet,
      simple: simple,
      download: true
    }) : null]
  });
};
export default MainPanel;
