// @ant-design/web3-assets/dist/lib/tokens/usdc.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.USDC = void 0;
var _web3Icons = require("@ant-design/web3-icons");
var _ethereum = require("../chains/ethereum");
var _solana = require("../solana");
var _jsxRuntime = require("react/jsx-runtime");
const USDC = exports.USDC = {
  name: 'USDC',
  symbol: 'USDC',
  decimal: 6,
  icon: /*#__PURE__*/(0, _jsxRuntime.jsx)(_web3Icons.UsdcColorful, {}),
  // Data Source:
  //   https://developers.circle.com/stablecoins/docs/usdc-on-main-networks
  availableChains: [
  // evm
  {
    chain: _ethereum.Mainnet,
    contract: '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48'
  }, {
    chain: _ethereum.Polygon,
    contract: '0x3c499c542cef5e3811e1192ce70d8cc03d5c3359'
  }, {
    chain: _ethereum.Arbitrum,
    contract: '0xaf88d065e77c8cc2239327c5edb3a432268e5831'
  }, {
    chain: _ethereum.Avalanche,
    contract: ''
  }, {
    chain: _ethereum.Optimism,
    contract: '0x0b2c639c533813f4aa9d7837caf62653d097ff85'
  },
  // solana
  {
    chain: _solana.Solana,
    contract: 'EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v'
  }]
};
