// @ant-design/web3/dist/lib/address/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Address = void 0;
var _react = _interopRequireWildcard(require("react"));
var _antd = require("antd");
var _classnames = _interopRequireDefault(require("classnames"));
var _hooks = require("../hooks");
var _useIntl = _interopRequireDefault(require("../hooks/useIntl"));
var _utils = require("../utils");
var _style = require("./style");
var _jsxRuntime = require("react/jsx-runtime");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
/**
 * Props for the Address component.
 */

const Address = props => {
  const {
    ellipsis,
    addressPrefix: addressPrefixProp,
    address,
    copyable = false,
    tooltip = true,
    format = false,
    children,
    locale
  } = props;
  const {
    getPrefixCls
  } = (0, _react.useContext)(_antd.ConfigProvider.ConfigContext);
  const {
    addressPrefix: addressPrefixContext
  } = (0, _hooks.useProvider)();
  const prefixCls = getPrefixCls('web3-address');
  const {
    wrapSSR,
    hashId
  } = (0, _style.useStyle)(prefixCls);
  const {
    messages
  } = (0, _useIntl.default)('Address', locale);
  const mergedFormat = (0, _react.useMemo)(() => {
    if (typeof format === 'function') {
      return format;
    }
    if (format) {
      return _utils.formatAddress;
    }
    return input => input;
  }, [format]);
  const isEllipsis = !!ellipsis;
  const {
    headClip = 6,
    tailClip = 4
  } = typeof ellipsis !== 'object' ? {
    headClip: 6,
    tailClip: 4
  } : ellipsis;
  if (!address) {
    return null;
  }
  const filledAddress = (0, _utils.fillWithPrefix)(address, addressPrefixProp, addressPrefixContext);
  const mergedTooltip = () => {
    if ( /*#__PURE__*/(0, _react.isValidElement)(tooltip) || typeof tooltip === 'string') {
      return tooltip;
    }
    if (tooltip) {
      return filledAddress;
    }
    return tooltip;
  };
  const formattedAddress = mergedFormat(filledAddress);
  return wrapSSR( /*#__PURE__*/(0, _jsxRuntime.jsx)(_antd.Space, {
    className: (0, _classnames.default)(prefixCls, hashId),
    children: /*#__PURE__*/(0, _jsxRuntime.jsx)(_antd.Typography.Text, {
      copyable: copyable ? {
        text: filledAddress,
        tooltips: [messages.copyTips, messages.copiedTips]
      } : false,
      children: /*#__PURE__*/(0, _jsxRuntime.jsx)(_antd.Tooltip, {
        title: mergedTooltip(),
        children: children ?? (isEllipsis ? `${filledAddress.slice(0, headClip)}...${filledAddress.slice(-tailClip)}` : formattedAddress)
      })
    })
  }));
};
exports.Address = Address;
